### PR TITLE
Farseen/01

### PR DESCRIPTION
**Pull request by itsfarseen**

### CODE REVIEW

File: DevNotes.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# Developer Notes
+
+### Development Workflow
+
+**Tools Required**:
+
+* `just`: https://github.com/casey/just \
```
Comment (nazrhom) [inline DevNotes.md:7]: I think we should stick to npm scripts, as they are usually the standard for node projects. If you really want to use `just` you can always install it as a dev dependency, and then provide both a justfile and npm scripts

File: DevNotes.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  We use `just` as our task runner as opposed to `package.json` scripts for readability. \
+  The tasks are described in `webext/justfile` and `wallet-lib/justfile`.
+* `entr`: https://eradman.com/entrproject/ \
```
Comment (nazrhom) [inline DevNotes.md:10]: Same here, we should install tools as devDependencies

File: DevNotes.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+`{browser}` can be `firefox` or `chrome`. If omitted, `firefox` is assumed.
+
+**Bundling**
+
+Run `just bundle {browser}` to create the extension package (crx/xpi).
+
+### Node Polyfills for ESBuild
+
+**History**
+* We are currently using https://github.com/imranbarbhuiya/esbuild-plugins-node-modules-polyfill \
```
Comment (nazrhom) [inline DevNotes.md:30]: This also needs to be added to package.json, otherwise it is not present when trying to build

File: webext/package.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  "name": "cardano-wallet",
+  "type": "module",
+  "version": "0.0.0",
+  "description": "A dev cardano wallet that works in all browsers",
+  "author": "MLabs <contact@mlabs.city>",
+  "license": "MIT",
+  "private": true,
+  "scripts": {},
+  "devDependencies": {
+    "crx": "^5.0.1",
```
Comment (nazrhom) [inline webext/package.json:11]:
```suggestion
    "crx": "^5.0.1",
    "esbuild-plugins-node-modules-polyfill": "^1.6.1",
```

File: DevNotes.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  Keep this in mind if issues with `eval()`s or failed imports of node
+  libraries start popping up.
+
+
+### CTL Dependency
+
+* We need to specify CTL as a dependency in `packages.dhall` as well as
+  `package.json`.
+  * The former is needed for importing CTL from Purescript.
+  * The latter is needed for `esbuild` to resolve the JS dependencies used by CTL for bundling.
```
Comment (nazrhom) [inline DevNotes.md:55]: Interesting! The alternative to this was to manually handle the list of all deps (see the old package.json) but I like this better. Still annoying we have to keep two references around :(

File: webext/package.json (hunk 4 above)
Comment (itsfarseen) [reply webext/package.json:11]: Good find @nazrhom. Thanks!. I had accidentally removed it and didn't notice it was gone as it was still present in my `node_modules` folder.

File: DevNotes.md (hunk 5 above)
Comment (itsfarseen) [reply DevNotes.md:55]: Yeah. Looks like this is is the least annoying option available as of now.

**Pull request merged**